performant-software/basira
# BASIRA #133 - Add #x of #y documents entered

### What this PR does

- Per #133:
  - Adds columns `documents_count` and `number_documents_visible` to `artworks` table
  - Adds `artwork_id` column to `documents` table
  - Also adds a migration to populate `documents_count` with current documents count for each artwork, and another to associate all `documents.artwork_id` columns with the correct artwork
  - Allows user to edit `number_documents_visible` from Artwork edit view
  - Shows "#x documents entered" or "#x of #y documents entered" on Artwork edit view sidebar (depending on whether or not `number_documents_visible` has been filled out)

### Status


### Steps to test

1. Visit https://basiraproject-staging.herokuapp.com/
2. Log in
3. Click the edit button by an artwork
4. Click the hamburger menu in the top left
5. Count the number of Document entries in that menu
6. Close the hamburger menu and verify that on the Artwork sidebar, it says "X documents entered" where X is the number you counted (unless it's 0)
7. Enter a value in the "Total documents visible" field and verify that the sidebar now says "X of Y documents entered" where Y is the value you entered
8. Click Save and verify that your changes persist
